src/Microsoft.AspNetCore.Sockets.Client/ServerSentEventsTransport.cs

```diff
@@ -0,0 +1,231 @@
+
+                    try
+                    {
+                        if (input.IsEmpty && result.IsCompleted)
+                        {
+                            _logger.LogDebug("Server-Sent Event Stream ended");
+                            break;
+                        }
+
+                        var parseResult = _parser.ParseMessage(input, out consumed, out examined, out var message);
+
+                        switch (parseResult)
+                        {
+                            case ServerSentEventsMessageParser.ParseResult.Completed:
+                                _application.Output.TryWrite(message);
+                                _parser.Reset();
+                                break;
+                            case ServerSentEventsMessageParser.ParseResult.Incomplete:
+                                if (result.IsCompleted)
+                                {
+                                    throw new FormatException("Incomplete message");
+                                }
+                                break;
+                        }
+                    }
+                    finally
+                    {
+                        pipelineReader.Advance(consumed, examined);
+                    }
+                }
+            }
+            finally
+            {
+                _transportCts.Cancel();
+            }
+        }
+
+        public async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Starting the send loop");
+
+            List<SendMessage> messages = null;
+            try
+            {
+                while (await _application.Input.WaitToReadAsync(cancellationToken))
+                {
+                    messages = new List<SendMessage>();
+                    while (!cancellationToken.IsCancellationRequested && _application.Input.TryRead(out SendMessage message))
+                    {
+                        messages.Add(message);
+                    }
+
+                    if (messages.Count > 0)
+                    {
+                        _logger.LogDebug("Sending {0} message(s) to the server using url: {1}", messages.Count, sendUrl);
+
+                        var request = new HttpRequestMessage(HttpMethod.Post, sendUrl);
+                        request.Headers.UserAgent.Add(DefaultUserAgentHeader);
+
+                        var memoryStream = new MemoryStream();
+
+                        var pipe = memoryStream.AsPipelineWriter();
+                        var output = new PipelineTextOutput(pipe, TextEncoder.Utf8);
+                        await WriteMessagesAsync(messages, output, MessageFormat.Binary);
+
+                        memoryStream.Seek(0, SeekOrigin.Begin);
+
+                        request.Content = new StreamContent(memoryStream);
+                        request.Content.Headers.ContentType = MediaTypeHeaderValue.Parse(MessageFormatter.GetContentType(MessageFormat.Binary));
```

Comment:
There's little reason for the C# client to use the text format, since it's only really for compatibility with Browsers. We can make it configurable, but file a bug if you want that and let's do it separately.